### PR TITLE
feat: dynamic definitions tree

### PR DESCRIPTION
**Pull request by ST-DDT**
Instead of making only our definitions/locale data accessible via `faker.definitions`, this PR allows devs to access any data, that are added to the locale itself.

Or in other words:
Previously `faker.definitions.business` returned `undefined` even though `faker.locales.en.buisness` is defined.

### CODE REVIEW

**Comment by ST-DDT** [comment]
Partially addresses #823 .

File: src/faker.ts [hunk 1]
```diff
@@ -98,43 +98,56 @@ export class Faker {
...
-          get: (): unknown =>
-            this.locales[this.locale][moduleName]?.[entryName] ??
-            this.locales[this.localeFallback][moduleName]?.[entryName],
-        });
-      }
-    }
+    };
+
+    // Returns the first LocaleDefinition[module][entry] in any locale
+    const resolveModuleData = (
```
Comment (Shinigami92) [inline src/faker.ts:130]: I would prefer to swap these `resolveModuleData` lines with `moduleLoader` lines cause in line 120 `resolveModuleData` is used but `moduleLoader` is not used in `resolveModuleData`

File: src/faker.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ export class Faker {
   localeFallback: UsableLocale;
 
   // Will be lazy init
-  readonly definitions: LocaleDefinition = {} as LocaleDefinition;
+  readonly definitions: LocaleDefinition = this.initDefinitions();
```
Comment (Shinigami92) [inline src/faker.ts:51]: Looks like it's not lazy init anymore, so I would just remove the comment

**Review by Shinigami92 — COMMENTED** [review]
To really understand what's going on in `initDefinitions` I would like to have a screenshare after my vacation with you, where I go through the code and you explain what I don't understand :slightly_smiling_face:

File: src/faker.ts (hunk 1 above)
Comment (ST-DDT) [reply src/faker.ts]: Fixed

File: src/faker.ts (hunk 2 above)
Comment (ST-DDT) [reply src/faker.ts]: Fixed

**Review by Shinigami92 — APPROVED** [review]
Tested it out locally with the new tests but without the changes in `src`
To me this is not a feature but a bugfix!

**Review by prisis — APPROVED** [review]

**Pull request merged**